File: dotnet_winform_simpleLotto/simpleLotto/service/LottoResultService.cs
using simpleLotto.data.model;
using simpleLotto.data.repository;
using simpleLotto.service.external;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace simpleLotto.service {
    public static class LottoResultService {
        public static async Task<LottoDrawResult> GetLottoDrawResult(int drwNo) {
            try {
                var result = await LottoDrawResultRepository.GetLottoDrawResult(drwNo);
                if (result == null) {
                    result = await LottoResultExternalService.GetLottoDrawResult(drwNo);
                    if (result != null) {
                        await LottoDrawResultRepository.AddLottoDrawResult(result);
                    }
                }
                return result;
            } catch (Exception ex) {
                Console.WriteLine(ex.Message);
                return null;
            }
        }
    }
}

File: flutter_lottoresult/lib/service/lotto_result_service.dart
import 'package:flutter_lottoresult/data/model/lotto_draw_result.dart';
import 'package:flutter_lottoresult/data/repository/lotto_draw_result_repository.dart';
import 'package:flutter_lottoresult/service/external/lotto_result_external_service.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

class LottoResultService {
  final LottoResultExternalService _externalService;
  final LottoDrawResultRepository _lottoDrawResultRepository;

  LottoResultService(this._externalService, this._lottoDrawResultRepository);

  Future<LottoDrawResult> getLottoDrawwResult(int drwNo) async {
    LottoDrawResult? result = _lottoDrawResultRepository
        .getLottoDrawResultByDrwNo(drwNo);
    if (result == null) {
      result = await _externalService.getDrawResult(drwNo);
      _lottoDrawResultRepository.insertLottoDrawResult(result);
    }
    return result;
  }
}

final lottoResultServiceProvider = Provider<LottoResultService>((ref) {
  final externalService = ref.watch(lottoResultExternalServiceProvider);
  final lottoDrawResultRepository = ref.watch(
    lottoDrawResultRepositoryProvider,
  );
  return LottoResultService(externalService, lottoDrawResultRepository);
});
